### PR TITLE
task: refactor and test volt module

### LINKED ISSUE #7: task: refactor and test volt module
### Description:

Refactor the volt module of the mainboard and add the missing unit tests.

### Implementation steps:

1. Change all the identifiers name following the embedded conventions
2. Implement unit tests
3. Fix small issues if any

### Dependencies:

None

### Additional notes:

_No response_

### PR DESCRIPTION
**Pull request by ale-giustina**
# Purpose
Refactoring of volt module

# Overview
Removed typedef on structs and enums, added doxygen comments and added unit tests.

# Guidance
The main files that have been affected are the header files in their dedicated folder, their implementation and the tests. Other files may have been modified marginally due to variable renaming.

Closes #7

### CODE REVIEW

File: Core/Inc/bms/volt/volt-api.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+#else // CONF_VOLTAGE_MODULE_ENABLE
+
+#define volt_api_init() (VOLT_OK)
+#define volt_api_get_values() (NULL)
+#define volt_api_get_min() (VOLT_MAX_VALUE)
+#define volt_api_get_max() (VOLT_MAX_VALUE)
+#define volt_api_get_avg() (VOLT_MAX_VALUE)
+#define volt_api_get_sum() (VOLT_VALUE_TO_VOLT(VOLT_MAX_VALUE))
+#define volt_api_cells_voltage_handle(payload) MAINBOARD_NOPE()
```
Comment (Tonidotpy) [inline Core/Inc/bms/volt/volt-api.h]:
```suggestion
#define volt_api_cells_voltage_handle(payload) EAGLETRT_API_NOP()
```

File: Core/Inc/bms/volt/volt-api.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+primary_hv_cells_voltage_stats_converted_t *volt_api_get_cells_voltage_stats_canlib_payload(size_t *byte_size);
+
+#else // CONF_VOLTAGE_MODULE_ENABLE
+
+#define volt_api_init() (VOLT_OK)
+#define volt_api_get_values() (NULL)
+#define volt_api_get_min() (VOLT_MAX_VALUE)
+#define volt_api_get_max() (VOLT_MAX_VALUE)
+#define volt_api_get_avg() (VOLT_MAX_VALUE)
+#define volt_api_get_sum() (VOLT_VALUE_TO_VOLT(VOLT_MAX_VALUE))
```
Comment (Tonidotpy) [inline Core/Inc/bms/volt/volt-api.h]: I don't know why I wrote this but these function should return the nominal voltage of the cells except the sum which should return the nominal voltage times the total number of cells (series) in the pack.

File: Core/Src/bms/volt-api.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    [VOLT_OK] = "ok",
+    [VOLT_NULL_POINTER] = "null pointer",
+    [VOLT_OUT_OF_BOUNDS] = "out of bounds"
+};
+
+EAGLETRT_STATIC char *volt_return_code_description[] = {
+    [VOLT_OK] = "executed successfully",
+    [VOLT_NULL_POINTER] = "attempt to dereference a null pointer",
+    [VOLT_OUT_OF_BOUNDS] = "attempt to access an invalid memory region"
+};
```
Comment (Tonidotpy) [inline Core/Src/bms/volt-api.c]: Missing `RC` prefix for return codes.

File: test/test_volt/test-volt.c [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+}
+
+void test_volt_api_get_cells_voltage_canlib_payload_offset_advances(void) {
+    volt_handler.cellboard_id = 0U;
+    volt_handler.offset = 0U;
+
+    (void)volt_api_get_cells_voltage_canlib_payload(NULL);
+
+    TEST_ASSERT_EQUAL_MESSAGE(3U, volt_handler.offset, "volt_api_get_cells_voltage_canlib_payload() did not advance offset");
+}
```
Comment (Tonidotpy) [inline test/test_volt/test-volt.c]: Testing the same behavior here, can be merged into a single test

File: test/test_volt/test-volt.c
Comment (Tonidotpy) [inline test/test_volt/test-volt.c:1]: Missing tests for `get_canlib_payload` functions with `NULL` byte size.

File: test/test_volt/test-volt.c [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    volt_handler.voltages[0U][0U] = 3.0F;
+    volt_handler.voltages[1U][0U] = 4.2F;
+
+    primary_hv_cells_voltage_stats_converted_t *payload = volt_api_get_cells_voltage_stats_canlib_payload(NULL);
+
+    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(0.0001F, 3.0F, payload->min, "volt_api_get_cells_voltage_stats_canlib_payload() wrong min");
+    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(0.0001F, 4.2F, payload->max, "volt_api_get_cells_voltage_stats_canlib_payload() wrong max");
+    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(0.0001F, 1.2F, payload->delta, "volt_api_get_cells_voltage_stats_canlib_payload() wrong delta");
+    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(0.01F, volt_api_get_avg(), payload->avg, "volt_api_get_cells_voltage_stats_canlib_payload() wrong avg");
+}
```
Comment (Tonidotpy) [inline test/test_volt/test-volt.c]: Can be merged

**Review by Tonidotpy — CHANGES_REQUESTED** [review]

File: Core/Inc/bms/volt/volt-api.h [hunk 6]
```diff
@@ -85,11 +86,11 @@ primary_hv_cells_voltage_stats_converted_t *volt_api_get_cells_voltage_stats_can
...
 #define volt_api_get_values() (NULL)
-#define volt_api_get_min() (VOLT_MAX_VALUE)
-#define volt_api_get_max() (VOLT_MAX_VALUE)
-#define volt_api_get_avg() (VOLT_MAX_VALUE)
-#define volt_api_get_sum() (VOLT_VALUE_TO_VOLT(VOLT_MAX_VALUE))
-#define volt_api_cells_voltage_handle(payload) MAINBOARD_NOPE()
+#define volt_api_get_min() ((VOLT_MAX_VALUE + VOLT_MIN_VALUE) / 2)
+#define volt_api_get_max() ((VOLT_MAX_VALUE + VOLT_MIN_VALUE) / 2)
+#define volt_api_get_avg() ((VOLT_MAX_VALUE + VOLT_MIN_VALUE) / 2)
+#define volt_api_get_sum() (VOLT_VALUE_TO_VOLT((VOLT_MAX_VALUE + VOLT_MIN_VALUE) / 2) * CELLBOARD_COUNT * CELLBOARD_SEGMENT_SERIES_COUNT)
```
Comment (Tonidotpy) [inline Core/Inc/bms/volt/volt-api.h]: I'd define a separate macro for the nominal voltage rather then using the average of the minimum and maximum voltage which is different.
Also the minimum and maximum voltages defined are the ones allowed by the bms which may differ from the real minimum and maximum voltages of the cells.

**Review by Tonidotpy — CHANGES_REQUESTED** [review]

File: Core/Inc/bms/volt/volt-api.h [hunk 7]
```diff
@@ -84,12 +84,14 @@ primary_hv_cells_voltage_stats_converted_t *volt_api_get_cells_voltage_stats_can
 
 #else // CONF_VOLTAGE_MODULE_ENABLE
 
+#define VOLT_NOMINAL_VALUE (4)
```
Comment (Tonidotpy) [inline Core/Inc/bms/volt/volt-api.h]: Not here, put it with the other two macros.

Comment (Tonidotpy) [reply Core/Inc/bms/volt/volt-api.h]: And that's not the nominal voltage of the cells either.

Comment (Tonidotpy) [inline Core/Inc/bms/volt/volt-api.h]: Using `V` instead of `VALUE` since it is a voltage.

File: Core/Inc/bms/volt/volt-api.h [hunk 8]
```diff
@@ -84,12 +84,14 @@ primary_hv_cells_voltage_stats_converted_t *volt_api_get_cells_voltage_stats_can
...
 #define volt_api_init() (VOLT_OK)
 #define volt_api_get_values() (NULL)
-#define volt_api_get_min() ((VOLT_MAX_VALUE + VOLT_MIN_VALUE) / 2)
-#define volt_api_get_max() ((VOLT_MAX_VALUE + VOLT_MIN_VALUE) / 2)
-#define volt_api_get_avg() ((VOLT_MAX_VALUE + VOLT_MIN_VALUE) / 2)
-#define volt_api_get_sum() (VOLT_VALUE_TO_VOLT((VOLT_MAX_VALUE + VOLT_MIN_VALUE) / 2) * CELLBOARD_COUNT * CELLBOARD_SEGMENT_SERIES_COUNT)
+#define volt_api_get_min() (VOLT_NOMINAL_VALUE)
+#define volt_api_get_max() (VOLT_NOMINAL_VALUE)
+#define volt_api_get_avg() (VOLT_NOMINAL_VALUE)
+#define volt_api_get_sum() (VOLT_VALUE_TO_VOLT(VOLT_NOMINAL_VALUE) * CELLBOARD_COUNT * CELLBOARD_SEGMENT_SERIES_COUNT)
```
Comment (Tonidotpy) [inline Core/Inc/bms/volt/volt-api.h]: Remove the `VOLT_VALUE_TO_VOLT` macro since the given value is already a voltage.

**Review by Tonidotpy — CHANGES_REQUESTED** [review]

**Review by Tonidotpy — APPROVED** [review]
Should be ok
